APP4080-Disaster-Management/APP4080-Disaster-Response-Coordination
# environments modified

**Description**
This PR introduces the "Disasters" feature to the app, enabling users to report, track, and respond to disaster events in real time. Key components include disaster data ingestion, live status updates, and geolocation-based alert systems for affected areas. This feature integrates with the roles for victims, responders, and coordinators to streamline communication and task assignment.

**Key Changes**
**Real-Time Disaster Reporting**

Added a reporting interface for users to submit disaster details, location, and severity level.
Implemented form validations to ensure complete and accurate reports.
**Mapping & Location Tracking**

Integrated Google Maps API to visualize disaster locations.
Users can pinpoint exact locations on the map, with automatic location detection if GPS is enabled.
Added map markers and overlays to indicate the severity of the disaster.
**Alert System**

Created an alerting mechanism to notify responders and coordinators immediately upon report submission.
Configured alerts to be role-specific, ensuring that only relevant users receive notifications.
**Dashboard Integration**

Displayed disaster reports on the responder and coordinator dashboards.
Created filters for location, type, and severity to help prioritize disaster responses.
**Related Changes**
New API Endpoint /api/disasters for posting and retrieving disaster data.
Unit Tests for the disaster reporting and alerting components.
Updated documentation to include instructions on using the "Disasters" feature.
Screenshots:
- 
![Screenshot (2)](https://github.com/user-attachments/assets/7f15a1a3-87e5-46d2-b133-2ec5aec6f30f)

**How to Test**
Navigate to the Disasters section in the app.
Submit a disaster report with details and confirm it appears on the map.
Check dashboards for updates and notifications.
**Notes**
Ensure .env contains a valid REACT_APP_GOOGLE_MAPS_API_KEY.
Make sure location permissions are enabled on testing devices for accurate geolocation.
**Checklist**
 Code is linted and follows the app's coding standards.
 API endpoint has been tested and validated.
 All tests pass successfully.
 Documentation is updated.
